Src/DependencyCollector/Shared.Tests/Implementation/ProfilerSqlProcessingTest.cs

```diff
@@ -361,6 +366,280 @@ public void OnEndLogsWarningWhenObjectPassedToEndWithoutCorrespondingBeginAsync(
             }
         }
 
+        [TestMethod]
+        public void FieldsForSqlConnectionAreCollectedCorrectly()
+        {
+            SqlConnection connection = GetSqlConnectionTest();
+            string actualResourceName = this.sqlConnectionProcessingProfiler.GetResourceName(connection);
+            Assert.AreEqual(ExpectedTarget, actualResourceName);
+
+            string actualTargetName = this.sqlConnectionProcessingProfiler.GetResourceTarget(connection);
+            Assert.AreEqual(ExpectedTarget, actualTargetName);
+
+            string actualCommandText = this.sqlConnectionProcessingProfiler.GetCommandName(connection);
+            Assert.AreEqual(ExpectedCommandTextForSqlConnection, actualCommandText);
+        }
+
+        [TestMethod]
+        public void OnEndAsyncSendsCorrectTelemetry_1ArgumentOverride()
+        {
+            var command = GetSqlCommandTestForStoredProc();
+            
+            Stopwatch stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            var returnTaskPassed = Task.Factory.StartNew(() => { });
+
+            var context = this.sqlCommandProcessingProfiler.OnBeginForOneParameter(command);
+            var objectReturned = this.sqlCommandProcessingProfiler.OnEndAsyncForOneParameter(context, returnTaskPassed, command);
+
+            stopwatch.Stop();
+
+            // wait for OnEnd async completion
+            Thread.Sleep(50);
```

Comment:
Is there something we can get a signal of completion from instead of sleeping?  Sometimes sleep dependent tests start failing when hardware changes or the framework changes.